Assets/Scripts/Enemy/EnemyService.cs

```diff
@@ -110,6 +110,17 @@ private void IncreaseDifficulty()
         public void SetEnemySpawning(bool setActive) => isSpawning = setActive;
 
         public void ReturnEnemyToPool(EnemyController enemyToReturn) => enemyPool.ReturnItem(enemyToReturn);
+
+        public void DestroyAllEnemies()
+        {
+            for(int i=0; i< enemyPool.pooledItems.Count; i++)
+            {
+                if (enemyPool.pooledItems[i].isUsed)
+                {
+                    enemyPool.pooledItems[i].Item.DestroyEnemy();
+                }
```

Comment:
We will be having a list of active enemies that we will loop over and destroy the view of each enemy, and return it back to pool. We will not be accessing and looping on pooledItems from somewhere external.